### PR TITLE
Make ESLint happy again

### PR DESCRIPTION
**Pull request by davidmehren**
This contains mostly more types at various places. We also excluded the migrations from eslint and fixed a bug in the note history updates

### CODE REVIEW

**Comment by davidmehren** [comment]
After an evening of testing, a few more bugs surfaced and were fixed. @InnayTool and I are now mostly sure that the realtime stuff still works.

File: src/lib/utils.ts [hunk 1]
```diff
@@ -4,6 +4,16 @@ import { logger } from './logger'
 import { Revision } from './models'
 import { realtime } from './realtime'
 
+/*
+Converts a map from string to something into a plain JSO for transmitting via a websocket
```
Comment (SISheogorath) [inline src/lib/utils.ts:8]: Typo? Shouldn't it be JSON?

Comment (SISheogorath) [inline src/lib/utils.ts:8]: Can we write it out? JSO was mistakenly considered a typo first, should be JS object.

Comment (davidmehren) [reply src/lib/utils.ts]: done

**Comment by SISheogorath** [comment]
@davidmehren feel free to press merge on this one :)

**Comment by davidmehren** [comment]
:rocket:

**Pull request merged**